### PR TITLE
SOLR-16711: Extract SolrCLI tool implementations into their own package and classes (Take 2)

### PR DESCRIPTION
**Pull request by epugh**
https://issues.apache.org/jira/browse/SOLR-16711

# Description

Second take on this.

# Solution

Please provide a short description of the approach taken to implement your solution.

# Tests

Please describe the tests you've developed or run to confirm this patch implements the feature or solves the problem.

# Checklist

Please review the following and check all that apply:

- [ ] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [ ] I have created a Jira issue and added the issue ID to my pull request title.
- [ ] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [ ] I have developed this patch against the `main` branch.
- [ ] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

**Comment by epugh** [comment]
Question?  Should we move `org.apache.solr.util.cli` up to `org.apache.solr.cli`?    And related...  Should we move `SolrCLI.java` into the `cli` package?

**Comment by HoustonPutman** [comment]
> Question? Should we move `org.apache.solr.util.cli` up to `org.apache.solr.cli`? And related... Should we move `SolrCLI.java` into the `cli` package?

Yes to both IMO

File: solr/packaging/test/test_export.bats [hunk 1]
```diff
@@ -29,7 +29,7 @@ teardown() {
 }
 
 @test "Check export command" {
-  run solr start -c -Dsolr.modules=sql
```
Comment (risdenk) [inline solr/packaging/test/test_export.bats:32]: Ummm why is this removed? This is checking a query so need sql module right?

File: solr/core/src/java/org/apache/solr/packagemanager/PackageManager.java [hunk 2]
```diff
@@ -356,7 +357,7 @@ private boolean deployPackage(
     List<String> previouslyDeployedOnCollections = deployResult.second();
 
     // Verify
-    boolean verifySuccess = true;
+    boolean verifySuccess;
     // Verify that package was successfully deployed
     verifySuccess =
```
Comment (risdenk) [inline solr/core/src/java/org/apache/solr/packagemanager/PackageManager.java:362]: Combine these lines?

**Review by risdenk — CHANGES_REQUESTED** [review]

File: solr/packaging/test/test_export.bats (hunk 1 above)
Comment (epugh) [reply solr/packaging/test/test_export.bats:32]: I don't think it uses the sql module..  it's just a straight up solr query, not a sql one....

**Review by gerlowskija — COMMENTED** [review]
I haven't had time to review all of the changes (5k LOC 🤯 ), but things LGTM from the brief skimming I was able to do.

I'd feel more confident in large refactors like this if the bats coverage was beefier to catch regressions in syntax/output, but that's probably an unrealistic expectation at this point and this is definitely a step in the right direction IMO.

**Comment by epugh** [comment]
> I haven't had time to review all of the changes (5k LOC 🤯 ), but things LGTM from the brief skimming I was able to do.
> 
> I'd feel more confident in large refactors like this if the bats coverage was beefier to catch regressions in syntax/output, but that's probably an unrealistic expectation at this point and this is definitely a step in the right direction IMO.

Yeah, I added two more bats tests, and actually the PDF that @bszabo97 uploaded on his work was super useful to me in thinking about these tests.     

I hope to add more bats tests as I look at each Tool in seperate PR's...

**Comment by epugh** [comment]
> Huge improvement to split this up. LGTM although have not studied every change.
> 
> Are there places where you did more than simply breaking classes out, where you'd like us to pay special attention?

I tried to restrain myself from "fixing" too many things.  The edits were mostly just responding to some intellij warnings..   I think as long as the changes to the bash/cmd scripts look good?   The bats tests run as do the unit tests, so I think it's actually not too much risk..

**Comment by janhoy** [comment]
Yea, and the script changes is basically java package name change. Looks good.
